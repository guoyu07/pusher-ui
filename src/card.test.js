import React from 'react';
import renderer from 'react-test-renderer';

import Card from './card';

describe('Card', () => {
  const defaultProps = { children: 'Hello, world!' };
  const createSnapshot = (props = defaultProps) => {
    expect(renderer.create(<Card {...props} />).toJSON()).toMatchSnapshot();
  };

  it('should render a card-like div by default', () => {
    createSnapshot();
  });

  it('should take any css props as usual', () => {
    createSnapshot({
      ...defaultProps,
      backgroundColor: 'hotpink',
      padding: 10,
    });
  });

  it('should take any props that the underlying component takes', () => {
    createSnapshot({
      ...defaultProps,
      children: [<div key="a">first</div>, <div key="b">second</div>],
      gutter: 24,
    });
  });
});
