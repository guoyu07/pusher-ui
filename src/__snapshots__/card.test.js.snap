// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render a card-like div by default 1`] = `
.glamor-0,
[data-glamor-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  padding: 1.25rem;
  background-color: #fff;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

<div
  className="glamor-0"
>
  Hello, world!
</div>
`;

exports[`Card should take any css props as usual 1`] = `
.glamor-0,
[data-glamor-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  padding: 10px;
  background-color: hotpink;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

<div
  className="glamor-0"
>
  Hello, world!
</div>
`;

exports[`Card should take any props that the underlying component takes 1`] = `
.glamor-0,
[data-glamor-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  padding: 1.25rem;
  background-color: #fff;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

<div
  className="glamor-0"
>
  <div>
    first
  </div>
  <div>
    second
  </div>
</div>
`;
